### ISSUE TITLE
Better Clouds Compatibility

### ISSUE THREAD

**Issue by Brickmaster1** [Talon396/EnhancedWeather#69]
https://modrinth.com/mod/better-clouds
Better clouds is arguably the most tasteful cloud improvement mod, it would be nice to have compat with it. Plus, better clouds has DH compatibility. Unfortunately, I do forsee that if this is a thing coding localized sections of clouds could be pretty difficult, but not so sure how modular better clouds is.

**Comment by TalonFloof** [comment]
Better Clouds does not have any way to add compatibility with other mods outside of it's own builtin compatibility code, which is understandable considering the lack of weather mods out there anyway. The only way Better Clouds could be compatible with Enhanced Weather is if the author(s) of Better Clouds added compatibility with Enhanced Weather.

On another note, I am aware of the cloud related issues and I've had serious thoughts of when I move this mod from 1.20.1 Fabric to 1.21 NeoForge rewriting the mod and making sure it has compatibility with Distant Horizons.

**Comment by Brickmaster1** [comment]
Oh, well I am a big fan of fabric, I actually enjoy that this mod uses it and is much better optimized and in depth than the alternatives. Also, you could just use mixins, right? When I want to add compatibility to a mod that does not provide APIs or anything, I just mixin into it. I do realize that this can be quite unstable and need patches a lot, but I mean if you mixin into a spot where you're fairly sure that it won't change in the future you should be good, right?

**Comment by TalonFloof** [comment]
Using mixins is possible, however the code of Better Clouds is not really built in mind to be expanded to another mod, plus I am highly unfamiliar with any of it's code as well which makes it more difficult to do it. In general, it's just something that doesn't really make any sense to add and doesn't really benefit the mod in any way.

Also to note, I am still debating on if I will stick with Fabric or switch to Neoforge. It's just something that I have considered, but the mod is more likely to stick with Fabric simply because I don't wanna rewrite it. It's just a potential idea i've thrown around. Should've probably clarified that.

**Comment by Qendolin** [comment]
I would be interested in adding an API for compat, but it might not be so easy.

**Comment by TalonFloof** [comment]
@Qendolin yeah, I have no idea how any of that would even work, I haven't even touched this mod in ages and honestly want to just completely rework what I have from the ground up. But idk, I have no idea if I'll even touch this mod anytime soon.
